src/database/parse-server.ts

```diff
@@ -0,0 +1,48 @@
+import { DatabaseConfig } from "./database-config";
+
+export default class ParseServer implements Database {
+  private static instance: Database;
+
+  private Parse: any;
+  private Event: any;
+
+  public static getInstance(Parse: any, AsyncStorage: any): Database {
+    if (this.instance == null) {
+      this.instance = new ParseServer(Parse, AsyncStorage);
+    }
+
+    return this.instance;
+  }
+
+  public async getEvents(
+    numResults: number,
+    pageNumber: number,
+    sortBy: string
+  ): Promise<any> {
+    const query = new this.Parse.Query(this.Event);
+
+    query.ascending(sortBy);
+    query.limit(numResults);
+    query.skip(numResults * (pageNumber - 1));
+    const results = await query.find();
+
+    return results;
+  }
+
+  private constructor(Parse: any, AsyncStorage: any) {
+    this.initialize(Parse, AsyncStorage);
+  }
+
+  private initialize(Parse: any, AsyncStorage: any): void {
+    this.Parse = Parse;
+    this.Parse.setAsyncStorage(AsyncStorage);
+
+    // TODO: Set config based on dev or prod environment
+    this.Parse.initialize(
+      DatabaseConfig.DEV.APP_ID,
+      DatabaseConfig.DEV.JAVASCRIPT_KEY
+    );
+    this.Parse.serverURL = DatabaseConfig.DEV.SERVER_URL;
+    this.Event = this.Parse.Object.extend("Event");
```

Comment:
should we have a DB config file for all the DB columns / tables?

Reply:
Good idea!